Grammar.md

```diff
@@ -0,0 +1,53 @@
+## Grammar support
+
+Many languages - all Slavic (Russian, Ukrainian, Polish, Bulgarian, etc), Finnic (Finnish, Estonian) and others - have [grammatical case feature](https://en.wikipedia.org/wiki/Grammatical_case) that could be supported in OSRM Text Instructions too.
+Originally street names are being inserted into instructions as they're in OSM map - in [nominative case](https://en.wikipedia.org/wiki/Nominative_case).
+To be grammatically correct, street names should be changed according to target language rules and instruction context before insertion.
```

Comment:
Are the various cases always regular and easy to determine from the nominative case? I noticed a few features have been tagged [`name:dative`](https://taginfo.openstreetmap.org/keys/name%3Adative) in OpenStreetMap – would it be desirable or necessary for OSRM to pass such tags along to OSRM Text Instructions?

Reply:
IMHO it's not deal of OSM - it's impossible to store all case variants in all languages for each name. Perhaps (CLDR)[http://cldr.unicode.org/] will be the better place (if will :wink: )